lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadata.php

```diff
@@ -329,6 +329,20 @@ class ClassMetadata implements ClassMetadataInterface
     public $parentClasses = array();
 
     /**
+     * READ-ONLY: Child class restrictions.
```

Comment:
the phpdoc should explain that if this list is empty, it means we don't check the classes of the children. to make a leaf node, use the explicit isLeaf field.